ctapipe/io/hdf5tableio.py

```diff
@@ -436,45 +436,54 @@ def read(self, table_name: str, container: Container, prefix=False):
             name of table to read from
         container : ctapipe.core.Container
             Container instance to fill
-        prefix: bool or str
+        prefix: bool, str or list
             Prefix that was added while writing the file.
             If True, the container prefix is taken into consideration, when
             comparing column names and container fields.
             If False, no prefix is used.
-            If a string is provided, it is used as prefix. This is äquivalent
-            to creating a container, changing its prefix and
-            using this container with prefix=True.
+            If a string is provided, it is used as prefix for all containers.
+            If a list is provided, the length needs to match th number
+            of containers.
         """
 
-        if prefix is True:
-            prefix = container.prefix
-        elif prefix is False:
-            prefix = None
+        if isinstance(containers, Container):
+            containers = (containers, )
+
+        if prefixes is False:
+            prefixes = ["" for container in containers]
+        elif prefixes is True:
+            prefixes = [container.prefix for container in containers]
+        elif isinstance(prefixes, str):
+            prefixes = [prefixes for container in containers]
+        assert len(prefixes) == len(containers)
 
         if table_name not in self._tables:
-            tab = self._setup_table(table_name, container, prefix)
+            tab = self._setup_table(table_name, containers, prefixes)
         else:
             tab = self._tables[table_name]
 
         row_count = 0
 
         while 1:
-
             try:
                 row = tab[row_count]
             except IndexError:
                 return  # stop generator when done
-
-            for colname in self._cols_to_read[table_name]:
-                if prefix and colname.startswith(prefix):
-                    colname_without_prefix = colname[len(prefix) + 1:]
-                else:
-                    colname_without_prefix = colname
-                container[colname_without_prefix] = self._apply_col_transform(
-                    table_name, colname, row[colname]
-                )
-
-            yield container
+            for container, prefix in zip(containers, prefixes):
+                for fieldname in container.keys():
+                    if prefix:
+                        colname = f"{prefix}_{fieldname}"
+                    else:
+                        colname = fieldname
+                    if colname not in self._cols_to_read[table_name]:
+                        continue
+                    container[fieldname] = self._apply_col_transform(
+                        table_name, colname, row[colname]
+                    )
+            if len(containers) == 1:
```

Comment:
This might be confusing for someone who passes in a list with one entry who then also expects a list of one entry back.

Reply:
Maybe just set a member variable storing if a single container was passed in or an iterable of containers